planner/core/logical_plan_builder.go

```diff
@@ -2474,3 +2488,34 @@ func getInnerFromParentheses(expr ast.ExprNode) ast.ExprNode {
 	}
 	return expr
 }
+
+// replaceVirtualColumn tries to replace a virtual generated column in given expression with its definition expression
+// stored in 'colMap'
+func replaceVirtualColumn(colMap map[int64]expression.Expression, expr expression.Expression) (expression.Expression, error) {
```

Comment:
This function looks like `ColumnSubstitute`, is it easy to adjust code to reuse it?

Reply:
aggre.
`Correlated column` can be held in other places. Use `ColumnSubstitute` seems a better choice.